sdks/python/apache_beam/testing/synthetic_pipeline.py

```diff
@@ -415,19 +418,24 @@ def get_range_tracker(self, start_position, stop_position):
       tracker = range_trackers.UnsplittableRangeTracker(tracker)
     return tracker
 
+  @staticmethod
+  def random_bytes(length):
+    """Return random bytes."""
+    return b''.join(
+        (struct.pack('B', random.getrandbits(8)) for _ in xrange(length)))
+
   def _gen_kv_pair(self, index):
-    r = np.random.RandomState(index)
-    rand = r.random_sample()
+    random.seed(index)
```

Comment:
How about we instantiate random.Random() instead of modifying the global state? This is important if we execute this generator concurrently. Not sure if we do.

Reply:
I'm pretty sure it's possible while having multiple range trackers and multiple workers.
I wrote a fix and ran the benchmark. The performance is a bit worse:
```
Python 2: 7.77728295326 (was 6.37386107445)
Python 3: 6.888783617999998 (was 5.6981477510000005)
```
However, we could create a cache for generators (instances of `random.Random`). Then we could create a generator for each range tracker, not for each index. This reduces the offset of instantiating multiple generators:
```
Python 2: 6.42752218246
Python 3: 5.789953311999998
```
WDYT?